python/ray/train/__init__.py

```diff
@@ -39,5 +45,6 @@
     "RunConfig",
     "ScalingConfig",
     "TrainingIterator",
+    "TrainingResult",
```

Comment:
Does this need to be aliased at the top level? Can we just keep it developer API in an internal module?

Reply:
I think DeveloperAPI should be in a public module. I agree, we should probably keep the top-level `train` namespace clean.

Maybe `train.trainer.TrainingResult`, but I'll just keep it in the internal module for now and we can see later.